src/parsemultipart.jl

```diff
@@ -1,17 +1,35 @@
+module MultiPartParsing
+
+import ..access_threaded, ..Request, ..Multipart, ..payload
+using ..Parsers
+
+export parse_multipart_form
+
 const CR_BYTE = 0x0d # \r
 const LF_BYTE = 0x0a # \n
 const DASH_BYTE = 0x2d # -
 const HTAB_BYTE = 0x09 # \t
 const SPACE_BYTE = 0x20
-const RETURN_BYTES = [CR_BYTE, LF_BYTE]
+const SEMICOLON_BYTE = UInt8(';')
+const CRLFCRLF = [CR_BYTE, LF_BYTE, CR_BYTE, LF_BYTE]
+
+"compare byte buffer `a` from index `i` to index `j` with `b` and check if they are byte-equal"
+function byte_buffers_eq(a, i, j, b)
+    l = 1
+    @inbounds for k = i:j
+        a[k] == b[l] || return false
+        l += 1
+    end
+    return true
+end
 
 """
     find_multipart_boundary(bytes, boundaryDelimiter; start::Int=1)
 
 Find the first and last index of the next boundary delimiting a part, and if
 the discovered boundary is the terminating boundary.
 """
-function find_multipart_boundary(bytes::AbstractVector{UInt8}, boundaryDelimiter::AbstractVector{UInt8}; start::Int=1)
+@inline function find_multipart_boundary(bytes::AbstractVector{UInt8}, boundaryDelimiter::AbstractVector{UInt8}; start::Int=1)
```

Comment:
I'm kind of amazed that inlining this function has any performance improvement.

Reply:
Oh, it doesn't. This was from an iteration where I was returning a SubString instead of indices and it could avoid the allocation if it was inlined.

Reply:
So should it have been removed?